File: backend/src/main/java/com/example/foodorderingapplication/services/Food/FoodService.java
package com.example.foodorderingapplication.services.Food;


import com.example.foodorderingapplication.db.entities.Category;
import com.example.foodorderingapplication.dto.FoodDetails;
import com.example.foodorderingapplication.dto.FoodDto;

import java.util.List;

public interface FoodService {

    List<FoodDto> getAllFoodByRestaurant(Long id);
    FoodDto createFood(Long id,FoodDetails foodDetails);
    FoodDto editFood(Long id,FoodDetails foodDetails);
    List<FoodDto> getFoodByCategory(Category category);
    void deleteFood(Long id);
}

File: frontend/FoodOrderingApp/src/app/services/food.service.ts
import { Injectable } from '@angular/core';
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {FoodDto} from "../dto/food-dto";
import {FoodDetails} from "../dto/food-details";
import {Foodcategory} from "../dto/foodcategory";

@Injectable({
  providedIn: 'root'
})
export class FoodService {

  foodURL:string;

  constructor(private http:HttpClient) {
    this.foodURL=`${environment.baseUrl}/food`
  }

  deleteFood(id : number) : Observable<FoodDto>
  {
    return this.http.delete<FoodDto>(`${this.foodURL}/${id}`)
  }

  getFoodByRestaurantId(id : number) : Observable<FoodDto[]>
  {
    return this.http.get<FoodDto[]>(`${this.foodURL}/restaurant/${id}`)
  }

  getFoodByCategory(category :Foodcategory): Observable<FoodDto[]>
  {
    return this.http.get<FoodDto[]>(`${this.foodURL}/${category}`)
  }

  addFood(foodDetails:FoodDetails,id:number) : Observable<FoodDto>
  {
    return this.http.post<FoodDto>(`${this.foodURL}/${id}`,foodDetails);
  }

  editFood(foodDetails:FoodDetails,id:number) : Observable<FoodDto>
  {
    return this.http.put<FoodDto>(`${this.foodURL}/${id}`,foodDetails)
  }
}
